src/main/java/seedu/address/model/person/weblink/WebLink.java

```diff
@@ -0,0 +1,97 @@
+package seedu.address.model.person.weblink;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.model.person.weblink.WebLinkUtil.FACEBOOK_MATCH_STRING;
+import static seedu.address.model.person.weblink.WebLinkUtil.FACEBOOK_TAG;
+import static seedu.address.model.person.weblink.WebLinkUtil.INSTAGRAM_MATCH_STRING;
+import static seedu.address.model.person.weblink.WebLinkUtil.INSTAGRAM_TAG;
+import static seedu.address.model.person.weblink.WebLinkUtil.LINKEDIN_MATCH_STRING;
+import static seedu.address.model.person.weblink.WebLinkUtil.LINKEDIN_TAG;
+import static seedu.address.model.person.weblink.WebLinkUtil.TWITTER_MATCH_STRING;
+import static seedu.address.model.person.weblink.WebLinkUtil.TWITTER_TAG;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+
+/**
+ * Represents a WebLink in the address book.
+ * Guarantees: immutable; name is valid as declared in {@link #isValidWebLink(String)}
+ * only accept twitter, facebook, linkedin and instagram links
+ * WebLinkTag defines the category of this webLink.
+ */
+public class WebLink {
+
+    public static final String MESSAGE_WEB_LINK_CONSTRAINTS = "Hi, only valid facebook,"
+            + " instagram, linkedin or twitter links will be accepted:)";
+    public static final String WEB_LINK_VALIDATION_REGEX = "^^.*(|instagram.com|linkedin.com|twitter.com"
```

Comment:
could use the regex to check instead.

This "weblink validation" doesn't make intuitive sense as it checks for instagram,linkedin,twitter rather than a weblink.

Consider a catchall weblink -> check for .com.

then further down have a "linkedin" regex, insta regex etc etc to split. If not, then go to default.

Reply:
wait doesnt instagram.com in the WEB_LINK_VALIDATION REGEX handle that .com issue you raised?